python/name-service/name_service/__init__.py

```diff
@@ -38,18 +50,32 @@ def get_year():
         r = requests.get(YEAR_ENDPOINT)
         return int(r.text)
 
+# shared resource service name
+myResource = Resource.create({"service.name": "name-python"})
 
-trace.set_tracer_provider(TracerProvider(
-    resource=Resource.create({SERVICE_NAME: "name-python"})
-))
+# tracing pipeline
+trace.set_tracer_provider(TracerProvider(resource=myResource))
```

Comment:
I think it makes more sense to create the `tracer_provider` with the resource and add the exporter to it before setting it with `set_tracer_provider`. It will save on setting then retrieving to add it.

Reply:
ok I've rearranged and then used that `tracer_provider` by name instead of getting it again - does this match what you had in mind?